sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -274,10 +305,39 @@ def text(self, encoding: Optional[str] = None) -> str:
 
         :param str encoding: The encoding to apply.
         """
+        # super().text detects charset based on self._body() which is compressed
+        # implement the decoding explicitly here
+        body = self.body()
+
+        ctype = self.headers.get(aiohttp.hdrs.CONTENT_TYPE, "").lower()
+        mimetype = aiohttp.helpers.parse_mimetype(ctype)
+
+        encoding = mimetype.parameters.get("charset")
+        if encoding:
+            try:
+                codecs.lookup(encoding)
+            except LookupError:
+                encoding = None
+        if not encoding:
+            if mimetype.type == "application" and (
+                    mimetype.subtype == "json" or mimetype.subtype == "rdap"
+            ):
+                # RFC 7159 states that the default encoding is UTF-8.
+                # RFC 7483 defines application/rdap+json
+                encoding = "utf-8"
+            elif body is None:
+                raise RuntimeError(
+                    "Cannot guess the encoding of " "a not yet read body"
+                )
+            else:
+                encoding = chardet.detect(body)["encoding"]
```

Comment:
Should this be `chardet.detect(body).get("encoding")`?

Reply:
This is the code from aiohttp. 

We want to keep it same because we want to have exactly same behavior of aiohttp. (I don't want users to notice we re-implement these logic in azure-core)

Reply:
`chardet.detect` always returns an `"encoding": something` key/value - even though it may have a very low confidence in its detection, it always makes a best effort.